### PR TITLE
feat: add linkedin provider

### PR DESCRIPTION
**Pull request by sh1luo**
Signed-off-by: wasabi <690898835@qq.com>

### CODE REVIEW

**Comment by sh1luo** [comment]
demo link: http://1.13.13.178/

File: web/src/ProviderEditPage.js [hunk 1]
```diff
@@ -74,7 +74,8 @@ class ProviderEditPage extends React.Component {
           {id: 'Facebook', name: 'Facebook'},
           {id: 'DingTalk', name: 'DingTalk'},
           {id: 'Weibo', name: 'Weibo'},
-          {id: 'Gitee', name: 'Gitee'}
+          {id: 'Gitee', name: 'Gitee'},
+          {id: 'Linkedin', name: 'Linkedin'},
```
Comment (hsluoyz) [inline web/src/ProviderEditPage.js:78]: All places should change to "LinkedIn", "I" is upper-case.

**Review by hsluoyz — CHANGES_REQUESTED** [review]
Another question is that can it get the profile URL for a LinkedIn user? 


Another bug: http://1.13.13.178/account , login admin, 123 , you will get:

image

**Comment by sh1luo** [comment]
@hsluoyz I fix the bug at `http://1.13.13.178/account `, now which won't report error in console. And I changed all `Linkedin` to `LinkedIn`

But I did not find the user profile url, because the suffix of the user profile url is customized and can be modified at will. I have not yet found the api to obtain this value.

For example, my profile url is: `https://www.linkedin.com/in/jikunliu/`

File: web/src/auth/Provider.js [hunk 2]
```diff
@@ -90,5 +96,7 @@ export function getAuthUrl(application, provider, method) {
     return `${WeiboAuthUri}?client_id=${provider.clientId}&redirect_uri=${redirectUri}&scope=${WeiboAuthScope}&response_type=code&state=${state}`;
   } else if (provider.type === "Gitee") {
     return `${GiteeAuthUri}?client_id=${provider.clientId}&redirect_uri=${redirectUri}&scope=${GiteeAuthScope}&response_type=code&state=${state}`;
+  } else if (provider.type === "Linkedin") {
```
Comment (hsluoyz) [inline web/src/auth/Provider.js:99]: I or i? Did you test the code?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by sh1luo** [comment]
@hsluoyz I'm sorry it was my mistake, now it's ok.

**Comment by hsluoyz** [comment]
@sh1luo plz attach the demo link for each PR.

**Review by hsluoyz — APPROVED** [review]

**Comment by sh1luo** [comment]
> demo link: http://1.13.13.178/

@hsluoyz

**Pull request merged**